ralink: add support for Linksys EA6350 v4

NullOranje opened the pull request:
Specifications:
- SoC: MT7621DAT (880MHz, 2 Cores)
- RAM: 128 MB
- Flash: 128 MB NAND
- Ethernet: 5x 1GiE MT7530
- WiFi: MT7603/MT7613
- USB: 1x USB 3.0

This is another MT7621 device, very similar to other Linksys EA7300
series devices.

Installation:

Upload the generated factory.bin image via the stock web firmware
updater.

Reverting to factory firmware:

Like other EA7300 devices, this device has an A/B router configuration
to prevent bricking.  Hard-resetting this device three (3) times will
put the device in failsafe (default) mode.  At this point, flash the
OEM image to itself and reboot.  This puts the router back into the 'B'
image and allows for a firmware upgrade.

Troubleshooting:

If the firmware will not boot, first restore the factory as described
above.  This will then allow the factory.bin update to be applied
properly.

Signed-off-by: Nick McKinney <nick@ndmckinney.net>

motolav commented:
If your kernel config changes aren't required for the device to boot then you should probably remove them from the commit

NullOranje commented:
> If your kernel config changes aren't required for the device to boot then you should probably remove them from the commit

Yeah, didn't mean to commit those.

Updated the entire commit; turns out, the device is almost identical to the 7300, so we can use much of the already defined configurations.

musashino205 commented on target/linux/ramips/mt7621/base-files/etc/board.d/02_network:
```diff
@@ -143,6 +143,7 @@ ramips_setup_macs()
 		label_mac=$(mtd_get_mac_binary factory 0x4)
 		;;
 	linksys,e5600|\
+    linksys,ea6350-v4|\
```
wrong indentation, also in the other base-files

NullOranje replied:
Updated.  Editor was using tab-to-spaces, which I didn't turn off.

musashino205 commented:
Please use the correct target name in the commit title, ```ramips:```.

NullOranje committed:
ramips: add support for Linksys EA6350 v4

Specifications:
- SoC: MT7621DAT (880MHz, 2 Cores)
- RAM: 128 MB
- Flash: 128 MB NAND
- Ethernet: 5x 1GiE MT7530
- WiFi: MT7603/MT7613
- USB: 1x USB 3.0

This is another MT7621 device, very similar to other Linksys EA7300
series devices.

Installation:

Upload the generated factory.bin image via the stock web firmware
updater.

Reverting to factory firmware:

Like other EA7300 devices, this device has an A/B router configuration
to prevent bricking.  Hard-resetting this device three (3) times will
put the device in failsafe (default) mode.  At this point, flash the
OEM image to itself and reboot.  This puts the router back into the 'B'
image and allows for a firmware upgrade.

Troubleshooting:

If the firmware will not boot, first restore the factory as described
above.  This will then allow the factory.bin update to be applied
properly.

Signed-off-by: Nick McKinney <nick@ndmckinney.net>

NullOranje commented:
> 

Updated commit title.  Never commit while tired.

hauke commented:
Thank you for the patch, I applied it to master in e0a574d4b73b6444d493866128cf3be3fae10892.
